File: plugins/openwrt/scripts.py
import os
from telegram import InlineKeyboardButton, InlineKeyboardMarkup
from .helpers import safe_callback_answer
from .connection import ssh_exec

async def wrt_scripts_list(update, context):
    query = update.callback_query
    await safe_callback_answer(query, "读取脚本列表...")
    script_dir = "/root/smart"
    res = ssh_exec(f"ls {script_dir}/*.sh 2>/dev/null")
    if not res:
        await query.edit_message_text(f"目录 {script_dir} 下没有找到脚本。", reply_markup=InlineKeyboardMarkup([[InlineKeyboardButton("🔙 返回", callback_data="wrt_main")]]))
        return
    scripts = res.splitlines()
    keyboard = []
    for s in scripts:
        name = os.path.basename(s)
        keyboard.append([InlineKeyboardButton(f"▶️ {name}", callback_data=f"wrt_run_{s}")])
    keyboard.append([InlineKeyboardButton("🔙 返回", callback_data="wrt_main")])
    await query.edit_message_text(f"📂 脚本列表 ({script_dir}):\n点击即可立即运行。", reply_markup=InlineKeyboardMarkup(keyboard))

async def wrt_run_script(update, context):
    query = update.callback_query
    script_path = query.data.replace("wrt_run_", "")
    await safe_callback_answer(query, "正在运行脚本...", show_alert=True)
    await query.edit_message_text(f"⏳ 正在执行: {script_path}\n请稍候...")
    res = ssh_exec(script_path)
    if res and len(res) > 3000:
        res = res[:3000] + "\n... (输出过长已截断)"
    result_text = f"✅ 执行完成: {script_path}\n\n📝 输出:\n{res}" if res else f"✅ 执行完成 (无输出)"
    keyboard = [[InlineKeyboardButton("🔙 返回脚本列表", callback_data="wrt_scripts_list")]]
    await query.edit_message_text(result_text, reply_markup=InlineKeyboardMarkup(keyboard))

File: pkg/openwrt/scripts.go
package openwrt

import (
	"fmt"
	"path/filepath"
	"strings"

	tele "gopkg.in/telebot.v3"
)

func HandleScriptsList(c tele.Context) error {
	c.Respond(&tele.CallbackResponse{Text: "读取脚本列表..."})

	scriptDir := "/root/smart"
	res, _ := SSHExec(fmt.Sprintf("ls %s/*.sh 2>/dev/null", scriptDir))

	menu := &tele.ReplyMarkup{}
	if strings.TrimSpace(res) == "" {
		menu.Inline(menu.Row(menu.Data("🔙 返回", "wrt_main")))
		return c.Edit(fmt.Sprintf("目录 %s 下没有找到脚本。", scriptDir), menu)
	}

	var rows []tele.Row
	scripts := strings.Split(strings.TrimSpace(res), "\n")
	for _, s := range scripts {
		if s == "" {
			continue
		}
		name := filepath.Base(s)
		rows = append(rows, menu.Row(menu.Data(fmt.Sprintf("▶️ %s", name), "wrt_run_script", s)))
	}
	rows = append(rows, menu.Row(menu.Data("🔙 返回", "wrt_main")))
	menu.Inline(rows...)

	return c.Edit(fmt.Sprintf("📂 脚本列表 (%s):\n点击即可立即运行。", scriptDir), menu)
}

func HandleRunScript(c tele.Context) error {
	parts := strings.SplitN(c.Callback().Data, "|", 2)
	if len(parts) < 2 {
		return c.Respond(&tele.CallbackResponse{Text: "Error: Invalid script path"})
	}
	scriptPath := parts[1]

	c.Respond(&tele.CallbackResponse{Text: "正在运行脚本...", ShowAlert: true})
	c.Edit(fmt.Sprintf("⏳ 正在执行: %s\n请稍候...", scriptPath))

	res, _ := SSHExec(scriptPath)
	if len(res) > 3000 {
		res = res[:3000] + "\n... (输出过长已截断)"
	}

	resultText := fmt.Sprintf("✅ 执行完成: %s\n\n📝 输出:\n%s", scriptPath, res)
	if res == "" {
		resultText = fmt.Sprintf("✅ 执行完成 (无输出): %s", scriptPath)
	}

	menu := &tele.ReplyMarkup{}
	menu.Inline(menu.Row(menu.Data("🔙 返回脚本列表", "wrt_scripts_list")))
	return c.Edit(resultText, menu)
}
